src/core/ext/filters/channel_idle/channel_idle_filter.cc

```diff
@@ -111,12 +113,21 @@ struct MaxAgeFilter::Config {
             .value_or(kDefaultMaxConnectionAgeGrace);
     // generate a random number between 1 - kMaxConnectionAgeJitter and
     // 1 + kMaxConnectionAgeJitter
-    const double multiplier =
-        rand() * kMaxConnectionAgeJitter * 2.0 / RAND_MAX + 1.0 -
-        kMaxConnectionAgeJitter;
+    struct BitGen {
+      Mutex mu;
+      absl::BitGen bit_gen ABSL_GUARDED_BY(mu);
+      double MakeUniformDouble(double min, double max) {
+        MutexLock lock(&mu);
+        return absl::Uniform(bit_gen, min, max);
+      }
+    };
+    static NoDestruct<PerCpu<BitGen>> bit_gen(PerCpuOptions().SetMaxShards(8));
+    const double multiplier = bit_gen->this_cpu().MakeUniformDouble(
```

Comment:
Is there any down-side to using the same jitter for both MAX_CONNECTION_AGE and MAX_CONNECTION_IDLE?  It seems like it should be okay, but I'm raising the question just to make sure we've considered it.

Reply:
Yeah I thought about this a little bit... I figured we might see some weird cases if they're decorrelated that I couldn't quite reason about, so I went with safe.